app/services/assignments/statistics.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module Assignments
+  # Service for calculating assignment statistics including criterion performance averages
+  class Statistics
+    def initialize(assignment)
+      @assignment = assignment
+    end
+
+    # Get performance statistics for all criteria
+    # Returns a hash with criterion as key and stats as value
+    def criterion_performance
+      return {} unless @assignment.rubric
+
+      @assignment.rubric.criteria.each_with_object({}) do |criterion, stats|
+        stats[criterion] = calculate_criterion_stats(criterion)
+      end
```

Comment:
This should be a stats object that can be called like: `stats.for(criterion)`. I don't really like working with hashes even if under the hood this is just a hash.